src/components/ModelCollapse.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Original file: https://github.com/swagger-api/swagger-ui/blob/3.x/src/core/components/model-collapse.jsx
+ * @prettier
+ */
+
+import React, { Component } from "react"
+
+export default class ModelCollapse extends Component {
+  constructor(props, context) {
+    super(props, context)
+
+    let { expanded, collapsedContent } = this.props
+
+    this.state = {
+      expanded: expanded,
+      collapsedContent: collapsedContent || ModelCollapse.defaultProps.collapsedContent
+    }
+  }
+
+  componentDidMount() {
+    const { hideSelfOnExpand, expanded, modelName } = this.props
+    if (hideSelfOnExpand && expanded) {
+      // We just mounted pre-expanded, and we won't be going back..
+      // So let's give our parent an `onToggle` call..
+      // Since otherwise it will never be called.
+      this.props.onToggle(modelName, expanded)
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.expanded !== nextProps.expanded) {
+      this.setState({ expanded: nextProps.expanded })
+    }
+  }
+
+  handleKeyDownToggle = (event) => {
+    if (event.nativeEvent.code === "Enter" || event.nativeEvent.code === "Space") {
+      this.toggleCollapsed()
+    }
+  }
+
+  toggleCollapsed = () => {
+    if (this.props.onToggle) {
+      this.props.onToggle(this.props.modelName, !this.state.expanded)
+    }
+
+    this.setState({
+      expanded: !this.state.expanded
+    })
+  }
+
+  render() {
+    const { title, classes } = this.props
+
+
+    return (
+      <div className={classes || ""}>
```

Comment:
Noticed that you changed main `span` into `div` - it won't cause any issues? 
Also, you removed function `this.onLoad` which was in original code -why?.